src/TxPool/Txpool.ts

```diff
@@ -0,0 +1,83 @@
+import Config from '../Config';
+import { Pool, ITxPoolTxDetails } from './Pool';
+
+export default class TxPool {
+    public config: Config;
+    public logger: any;
+    public pool: any = [];
+    public subs: any = {};
+  
+    constructor(config: Config) {
+      this.config = config;
+      this.pool = new Pool();
+    }
+
+    public running () {
+        return !this.pool.isEmpty() || this.subs.pending || this.subs.latest;
+    }
+
+    public async start () {
+        if (this.running()) {
+            await this.stop();
+        }
+        await this.watchPending();
+        await this.watchLatest();
+    }
+
+    public async stop () {
+        if (this.subs.pending) {
+            await this.subs.pending.stopWatching();
+        }
+        if (this.subs.latest) {
+            await this.subs.latest.stopWatching();
+        }
+        this.pool.wipe();
+        this.subs = {};
+    }
+
+    public async watchPending () {
+        this.subs.pending = await this.config.web3.eth.filter('pending');
+        this.subs.pending.watch( async (err: any, res: any) => {
+            if (err) {
+                return this.config.logger.error(err);
+            }
+
+            if (this.pool.preSet(res)){
+                await this.config.web3.eth.getTransaction(res,
```

Comment:
This is not awaitable. 
Maybe create getTransaction wrapper in Utils.ts similarly to other functions. Then can await and reduce the clutter in the code.